crates/dapf/src/http_client.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2024 Cloudflare, Inc. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+use std::{borrow::Cow, env, io::Cursor, path::Path};
+
+use anyhow::{anyhow, bail, Context};
+use daphne::messages::{decode_base64url_vec, HpkeConfigList};
+use daphne_service_utils::http_headers;
+use prio::codec::Decode;
+use reqwest::{Client, Identity};
+use url::Url;
+use webpki::{EndEntityCert, ECDSA_P256_SHA256};
+use x509_parser::pem::Pem;
+
+use crate::response_to_anyhow;
+
+pub struct HttpClient {
+    using_mtls: bool,
+    inner: HttpClientInner,
+}
+
+#[allow(clippy::large_enum_variant)]
+enum HttpClientInner {
+    NoReuse {
+        identity: Option<Identity>,
+        ssl_key_log_file_enabled_tls: Option<rustls::ClientConfig>,
+    },
+    Reuse(Client),
+}
+
+fn load_identity() -> anyhow::Result<Option<Identity>> {
+    // LEADER_ variables left here for backwards compatibility
+    const TLS_CLIENT_CERT_VAR: [&str; 2] = ["LEADER_TLS_CLIENT_CERT", "TLS_CLIENT_CERT"];
+    const TLS_CLIENT_KEY_VAR: [&str; 2] = ["LEADER_TLS_CLIENT_KEY", "TLS_CLIENT_KEY"];
+
+    let leader_tls_client_cert = TLS_CLIENT_CERT_VAR
+        .iter()
+        .find_map(|key| env::var(key).ok());
```

Comment:
nit: Probably better to check for the non-legacy variable first, i.e., "TLS_CLIENT_CERT" should be preferred over "LEADER_TLS_CLIENT_CERT" if both exist